frappe/model/base_document.py

```diff
@@ -156,14 +177,20 @@ def getone(self, key, filters=None):
 		return self.get(key, filters=filters, limit=1)[0]
 
 	def set(self, key, value, as_value=False):
-		if key in self.ignore_in_setter:
+		if key in self._ignore_in_setter:
 			return
 
-		if isinstance(value, list) and not as_value:
-			self.__dict__[key] = []
-			self.extend(key, value)
-		else:
-			self.__dict__[key] = value
+		if not as_value:
+			if isinstance(value, list):
+				self.__dict__[key] = []
+				self.extend(key, value)
+				return
+
+			# Did you mean empty list?
+			if value is None and key in self._table_fieldnames:
+				value = []
```

Comment:
We could go a step further here, and only call `self.extend` if they key is in `_table_fieldnames` as opposed to anything that is a list. The benefit of this would be being able to set lists as values without them necessarily being children.

Reply:
[In action](https://github.com/resilient-tech/frappe/compare/perf-doc-get2...resilient-tech:perf-init-child)

Reply:
I have added this to current PR itself.